src/shared/utilities/timeoutUtils.ts

```diff
@@ -77,3 +77,28 @@ export class Timeout {
         }
     }
 }
+
+/**
+ * Invokes `fn()` until it returns a truthy value.
+ *
+ * @param fn  Function whose result is checked
+ * @param opt.timeout  Timeout in ms (default: 5000)
+ * @param opt.interval  Interval in ms between fn() checks (default: 500)
+ *
+ * @returns Result of `fn()`, or `undefined` if timeout was reached.
+ */
+export async function waitUntil<T>(
+    fn: () => Promise<T>,
+    opt: { timeout: number; interval: number } = { timeout: 5000, interval: 500 }
+): Promise<T | undefined> {
+    for (let i = 0; true; i++) {
```

Comment:
I don't like the `true` here, can you just bump the if statement wrapping the `return undefined` here and return undefined outside of the for loop?

Reply:
Why?

Reply:
I can understand this if you have a `break` that doesn't rely on the value of `i` but you have a built-in comparison against the `i` value later in the loop, which is what this space should be used for. I don't have a built-in coding pattern or practice from a senior principal engineer on hand, but it just feels unconventional for very little reason (I guess this allows for one last run at the `fn()` statement prior to returning `undefined`).

Reply:
Unconventional how? It is common that for-loops have conditions other than `i < x`, as well as `break` and `return` statements.